samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/MainApplication.kt

```diff
@@ -19,31 +15,35 @@ import kotlinx.coroutines.runBlocking
 class MainApplication : Application() {
 
     @Inject
-    lateinit var dataStore: DataStore<Preferences>
+    lateinit var preferences: PreferenceRepository
+
     override fun onCreate() {
         super.onCreate()
-        val (trackingApiUrl, siteId, apiKey) = runBlocking {
-            return@runBlocking try {
-                val data = dataStore.data.first()
-                listOf(
-                    data[TRACK_API_URL_KEY],
-                    data[SITE_ID],
-                    data[API_KEY]
-                )
-            } catch (e: Exception) {
-                listOf(null, BuildConfig.SITE_ID, BuildConfig.API_KEY)
-            }
+        val configuration = runBlocking {
+            preferences.getConfiguration().first()
         }
 
         CustomerIO.Builder(
-            siteId = siteId ?: BuildConfig.SITE_ID,
-            apiKey = apiKey ?: BuildConfig.API_KEY,
+            siteId = configuration.siteId,
+            apiKey = configuration.apiKey,
             appContext = this
         ).apply {
-            trackingApiUrl?.let { setTrackingApiURL(trackingApiUrl = it) }
+            configuration.trackUrl?.let {
+                setTrackingApiURL(trackingApiUrl = it)
+            }
+            setBackgroundQueueSecondsDelay(configuration.backgroundQueueSecondsDelay)
+            setBackgroundQueueMinNumberOfTasks(configuration.backgroundQueueMinNumTasks)
+            if (configuration.debugMode) {
+                setLogLevel(CioLogLevel.DEBUG)
+            } else {
+                setLogLevel(CioLogLevel.ERROR)
+            }
+            autoTrackDeviceAttributes(configuration.trackDeviceAttributes)
+            autoTrackScreenViews(configuration.trackScreen)
+
             addCustomerIOModule(ModuleMessagingInApp())
             addCustomerIOModule(ModuleMessagingPushFCM())
-            setLogLevel(CioLogLevel.DEBUG)
+
             build()
         }
```

Comment:
I like the idea of trying to make the sample apps source code helpful for customers to read and learn how to use our SDK in their app. 

The `configuration` object is an object that is unique to our sample apps and not relevant to customers. Therefore, I think that it would be ideal if the code that configures the SDK via the `configuration` object is abstracted away from this code block. 

Here is some code to explain the idea:

```kt
        // Initialize the Customer.io SDK
        CustomerIO.Builder(siteId: siteId, apiKey: apiKey, appContext: this).let { builder -> 
            // Configure the Customer.io SDK through the builder: 
			builder.setLogLevel(CioLogLevel.DEBUG) // For all of our sample apps, we prefer to set debug logs to make our internal testing easier. You may not need to do this.

            // This line of code is internal to Customer.io for testing purposes. Do not add this code to your app.
            configureSdkFromAppSettings(configuration, builder)

			// Finally, build the builder to finish initialization 
			builder.build()
        }
```

The idea is that we abstract away the `configuration` object modifications into a separate function `configureSdkFromAppSettings`. 

This code is similar to [what is in the FCM iOS sample app](https://github.com/customerio/customerio-ios/blob/a0224143b12b9bcdb1548965f35fbd38c8b6d61a/Apps/CocoaPods-FCM/src/AppDelegate.swift#L18-L24) and [the java sample app](https://github.com/customerio/customerio-android/blob/3aa5addd752a748eba53be61baae57e619d11540/samples/java_layout/src/main/java/io/customer/android/sample/java_layout/SampleApplication.java#L35-L52). 

For more details, you can see [a similar comment I left in java sample app](https://github.com/customerio/customerio-android/pull/198#discussion_r1188786852).

Reply:
Let me see what can be done about this.

Reply:
In case it helps, I have moved all SDK calls to a repository [like this](https://github.com/customerio/customerio-android/pull/203/files#diff-d69d87bddf0123142bdd66da3e46acad1ded0fa900f1877c5617d23bfca5276b) in Java sample.